src/Adapter/Language/LanguageCopier.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Language;
+
+use PrestaShop\PrestaShop\Core\Language\Copier\LanguageCopierConfigInterface;
+use PrestaShop\PrestaShop\Core\Language\Copier\LanguageCopierInterface;
+use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class LanguageCopier responsible for copying a language into another language
+ */
+final class LanguageCopier implements LanguageCopierInterface
+{
+    /**
+     * @var LanguageDataProvider
+     */
+    private $languageDataProvider;
+
+    /**
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param LanguageDataProvider $languageDataProvider
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        LanguageDataProvider $languageDataProvider,
+        TranslatorInterface $translator
+    ) {
+        $this->languageDataProvider = $languageDataProvider;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function copyLanguage(LanguageCopierConfigInterface $config)
+    {
+        $languageFiles = $this->languageDataProvider->getFilesList(
+            $config->getLanguageFrom(),
+            $config->getThemeFrom(),
+            $config->getLanguageTo(),
+            $config->getThemeTo(),
+            true
```

Comment:
can we remove this boolean attribute and create a new function instead, with a more meaningful name?